justfile

```diff
@@ -68,3 +69,33 @@ token WORKSPACE="workspace":
     . scripts/setup_functions
     dev_setup
     python dev-token.py "{{ WORKSPACE }}"
+
+
+
+# docker-compose bails if it cannot find this file, even though we don't
+# actually need it to build, so ensure it exists.
+# ensure .env file
+env:
+    @test -f .env || cp dotenv-sample.env .env
+
+# build docker image env=dev|prod
+docker-build env="dev": env
+    #!/usr/bin/env bash
+    # enable modern docker build features
+    export DOCKER_BUILDKIT=1
+    export COMPOSE_DOCKER_CLI_BUILD=1
+    # set build args
+    export USERID=$(id -u)
+    export BUILD_DATE=$(date +'%y-%m-%dT%H:%M:%S.%3NZ')
```

Comment:
macOS ships with BSD `date`, so this line produces `21-07-30T14:57:15.3NZ`, is that expected?

Reply:
it most certainly is not, hmm. we can probably figure out something that works on both

but the actual prod build will be done from GHA, so it may not matter too much

Reply:
I've dropped the .%3N, which wasn't really needed, and added -u, to force UTC.